Version 22.0.0 not started in dev mode and build mode

### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

core

### Describe the bug

Version 22.0.0 not started in dev mode and build mode

### Version

22.0.0

### Expected behavior

Version 22.0.0 not started in dev mode and build mode:

Updating the configuration and installing your custom providers, if any. Please wait.
2023-07-12 12:06:19,472 WARN  [io.quarkus.arc.processor.BeanArchives] (build-32) Failed to index org.keycloak.models.map.storage.jpa.JpaMapStorageProviderFactory: Class does not exist in ClassLoader QuarkusClassLoader:Deployment Class Loader: PROD@13d186db
2023-07-12 12:06:19,491 INFO  [io.quarkus.arc.processor.IndexClassLookupUtils] (build-32) Class for name: org.keycloak.models.map.storage.jpa.JpaMapStorageProviderFactory was not found in Jandex index. Please ensure the class is part of the index.
ERROR: Failed to run 'build' command.
ERROR: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.deployment.pkg.steps.JarResultBuildStep#buildRunnerJar threw an exception: java.lang.IllegalStateException: java.nio.file.FileSystemException: C:\APP\keycloak\lib\quarkus\generated-bytecode.jar: The process cannot access the file because it is being used by another process
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:918)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
        at java.base/java.lang.Thread.run(Thread.java:833)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.nio.file.FileSystemException: C:\APP\keycloak\lib\quarkus\generated-bytecode.jar: The process cannot access the file because it is being used by another process
        at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:92)
        at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
        at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
        at java.base/sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:275)
        at java.base/sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:105)
        at java.base/java.nio.file.Files.delete(Files.java:1152)
        at io.quarkus.bootstrap.util.IoUtils.createOrEmptyDir(IoUtils.java:114)
        at io.quarkus.deployment.pkg.steps.JarResultBuildStep.buildThinJar(JarResultBuildStep.java:600)
        at io.quarkus.deployment.pkg.steps.JarResultBuildStep.buildRunnerJar(JarResultBuildStep.java:223)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
        ... 6 more

ERROR: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.deployment.pkg.steps.JarResultBuildStep#buildRunnerJar threw an exception: java.lang.IllegalStateException: java.nio.file.FileSystemException: C:\APP\keycloak\lib\quarkus\generated-bytecode.jar: The process cannot access the file because it is being used by another process
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:918)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
        at java.base/java.lang.Thread.run(Thread.java:833)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.nio.file.FileSystemException: C:\APP\keycloak\lib\quarkus\generated-bytecode.jar: The process cannot access the file because it is being used by another process
        at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:92)
        at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
        at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
        at java.base/sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:275)
        at java.base/sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:105)
        at java.base/java.nio.file.Files.delete(Files.java:1152)
        at io.quarkus.bootstrap.util.IoUtils.createOrEmptyDir(IoUtils.java:114)
        at io.quarkus.deployment.pkg.steps.JarResultBuildStep.buildThinJar(JarResultBuildStep.java:600)
        at io.quarkus.deployment.pkg.steps.JarResultBuildStep.buildRunnerJar(JarResultBuildStep.java:223)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
        ... 6 more

ERROR: java.nio.file.FileSystemException: C:\APP\keycloak\lib\quarkus\generated-bytecode.jar: The process cannot access the file because it is being used by another process
ERROR: C:\APP\keycloak\lib\quarkus\generated-bytecode.jar: The process cannot access the file because it is being used by another process
For more details run the same command passing the '--verbose' option. Also you can use '--help' to see the details about the usage of the particular command.

### Actual behavior

Version 22.0.0 not started in dev mode and build mode

### How to Reproduce?

Version 22.0.0 not started in dev mode and build mode

### Anything else?

_No response_

Launch kc build on fresh environment:

kc build
Updating the configuration and installing your custom providers, if any. Please wait.
2023-07-12 13:32:03,171 WARN  [io.quarkus.arc.processor.BeanArchives] (build-41) Failed to index org.keycloak.models.map.storage.jpa.JpaMapStorageProviderFactory: Class does not exist in ClassLoader QuarkusClassLoader:Deployment Class Loader: PROD@42561fba
2023-07-12 13:32:03,186 INFO  [io.quarkus.arc.processor.IndexClassLookupUtils] (build-41) Class for name: org.keycloak.models.map.storage.jpa.JpaMapStorageProviderFactory was not found in Jandex index. Please ensure the class is part of the index.
ERROR: Failed to run 'build' command.
ERROR: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.deployment.pkg.steps.JarResultBuildStep#buildRunnerJar threw an exception: java.lang.IllegalStateException: java.nio.file.FileSystemException: E:\APPS\keycloak\lib\quarkus\generated-bytecode.jar: The process cannot access the file because it is being used by another process
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:918)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
        at java.base/java.lang.Thread.run(Thread.java:833)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.nio.file.FileSystemException: E:\APPS\keycloak\lib\quarkus\generated-bytecode.jar: The process cannot access the file because it is being used by another process
        at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:92)
        at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
        at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
        at java.base/sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:275)
        at java.base/sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:105)
        at java.base/java.nio.file.Files.delete(Files.java:1152)
        at io.quarkus.bootstrap.util.IoUtils.createOrEmptyDir(IoUtils.java:114)
        at io.quarkus.deployment.pkg.steps.JarResultBuildStep.buildThinJar(JarResultBuildStep.java:600)
        at io.quarkus.deployment.pkg.steps.JarResultBuildStep.buildRunnerJar(JarResultBuildStep.java:223)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
        ... 6 more

ERROR: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.deployment.pkg.steps.JarResultBuildStep#buildRunnerJar threw an exception: java.lang.IllegalStateException: java.nio.file.FileSystemException: E:\APPS\keycloak\lib\quarkus\generated-bytecode.jar: The process cannot access the file because it is being used by another process
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:918)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
        at java.base/java.lang.Thread.run(Thread.java:833)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.nio.file.FileSystemException: E:\APPS\keycloak\lib\quarkus\generated-bytecode.jar: The process cannot access the file because it is being used by another process
        at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:92)
        at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
        at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
        at java.base/sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:275)
        at java.base/sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:105)
        at java.base/java.nio.file.Files.delete(Files.java:1152)
        at io.quarkus.bootstrap.util.IoUtils.createOrEmptyDir(IoUtils.java:114)
        at io.quarkus.deployment.pkg.steps.JarResultBuildStep.buildThinJar(JarResultBuildStep.java:600)
        at io.quarkus.deployment.pkg.steps.JarResultBuildStep.buildRunnerJar(JarResultBuildStep.java:223)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
        ... 6 more

ERROR: java.nio.file.FileSystemException: E:\APPS\keycloak\lib\quarkus\generated-bytecode.jar: The process cannot access the file because it is being used by another process
ERROR: E:\APPS\keycloak\lib\quarkus\generated-bytecode.jar: The process cannot access the file because it is being used by another process
For more details run the same command passing the '--verbose' option. Also you can use '--help' to see the details about the usage of the particular command.

--- a/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/Build.java
+++ b/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/Build.java
@@ -75,10 +75,9 @@ public final class Build extends AbstractCommand implements Runnable {
         println(spec.commandLine(), "Updating the configuration and installing your custom providers, if any. Please wait.");
 
         try {
-            beforeReaugmentationOnWindows();
-
             configureBuildClassLoader();
 
+            beforeReaugmentationOnWindows();
             QuarkusEntryPoint.main();
 
             if (!isDevMode()) {

--- a/quarkus/runtime/src/main/resources/application.properties
+++ b/quarkus/runtime/src/main/resources/application.properties
@@ -41,6 +41,8 @@ quarkus.http.non-application-root-path=/${quarkus.http.root-path}
 
 # Disable specific categories from logs
 quarkus.log.category."io.quarkus.config".level=off
+quarkus.log.category."io.quarkus.arc.processor.BeanArchives".level=off
+quarkus.log.category."io.quarkus.arc.processor.IndexClassLookupUtils".level=off
 quarkus.log.category."io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor".level=warn
 
 # Set default directory name for the location of the transaction logs

